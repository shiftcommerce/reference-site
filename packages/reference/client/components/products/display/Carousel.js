import { Component } from 'react'
import { Carousel } from 'react-responsive-carousel'

class CarouselComponent extends Component {
  render () {
    const {
      product
    } = this.props

    const imageUrls = product.asset_files.map(a => a.s3_url)

    return (
      <Carousel showStatus={false} useKeyboardArrows>
        {imageUrls.map((imageUrl, index) => (
          <div key={index}>
            <img className='c-slider-image' src={imageUrl} />
          </div>
        ))}
      </Carousel>
    )
  }
}

export default CarouselComponent
