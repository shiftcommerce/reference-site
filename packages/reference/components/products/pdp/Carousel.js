import { Component } from 'react'
import Slider from 'react-slick'

import Image from '../../../objects/Image'

class Carousel extends Component {
  render () {
    const {
      product
    } = this.props

    const settings = {
      fade: true,
      arrows: true,
      dots: true,
      infinite: true,
      speed: 500,
      slidesToShow: 1,
      slidesToScroll: 1
    }

    const imageUrls = product.asset_files.map(a => a.s3_url)

    return (
      <div>
        <Slider {...settings}>
          {imageUrls.map(imageUrl => (
            <div key={imageUrl}>
              <Image className='c-slider-image' src={imageUrl} />
            </div>
          ))}
        </Slider>
      </div>
    )
  }
}

export default Carousel
