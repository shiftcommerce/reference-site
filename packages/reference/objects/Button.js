// Libraries
import { Component } from 'react'
import classNames from 'classnames'

class Button extends Component {
  render () {
    let {
      status,
      label,
      size,
      disabled,
      className,
      ...otherProps
    } = this.props

    disabled = disabled || false
    status = status || 'primary'
    size = size || 'lrg'

    return (
      <button
        className={classNames(className, 'o-button', `o-button--${status}`, `o-button--${size}`, {'o-button--disabled': disabled})}
        tabIndex='0'
        role='button'
        disabled={disabled}
        {...otherProps}
      >
        { label }
      </button>
    )
  }
}

export default Button
