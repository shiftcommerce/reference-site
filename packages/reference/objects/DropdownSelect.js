// Libraries
import { Component } from 'react'
import classNames from 'classnames'

// HOC
import { withValidationMessage } from './withValidationMessage'

class DropdownSelect extends Component {
  constructor (props) {
    super(props)

    this.triggerBlur = this.triggerBlur.bind(this)
    this.triggerChange = this.triggerChange.bind(this)
  }

  renderOptions (options) {
    return (
      options && options.map((option, idx) =>
        <option role='option' key={idx} value={option.value || option} aria-setsize={options.length} aria-posinset={idx + 1} >{ option.title || option }</option>
      )
    )
  }

  triggerChange (event) {
    const { formName, name, onChange } = this.props
    if (onChange) {
      onChange(event, formName, name, event.target.value)
    }
  }

  triggerBlur (event) {
    const { formName, name, rules, onBlur } = this.props
    if (onBlur) {
      onBlur(event, formName, name, event.target.value, rules)
    }
  }

  renderDropdown () {
    let {
      name,
      value,
      prompt,
      options,
      required,
      className,
      validationMessage
    } = this.props

    let validationErrorPresent = (validationMessage !== undefined) && (validationMessage !== '')

    return (
      <select
        className={classNames('o-form__input-block o-form__input-field', className, { 'o-form__input-field__error': validationErrorPresent })}
        id={name}
        required={required}
        value={value}
        name={name}
        role='listbox'
        onChange={this.triggerChange}
        onBlur={this.triggerBlur}
      >
        <option role='option' value='' aria-setsize={options.length} aria-posinset='0'>{ prompt }</option>
        { this.renderOptions(options) }
      </select>
    )
  }

  renderLabel () {
    const { label, name, required } = this.props

    if (label) {
      return (
        <label htmlFor={name} className='o-form__label'>{label}{required && ' *'}</label>
      )
    }
  }

  render () {
    return (
      <div className={classNames('o-form__input-group')}>
        { this.renderLabel() }
        { this.renderDropdown() }
        { this.props.renderValidationMessage() }
      </div>
    )
  }
}

export default withValidationMessage()(DropdownSelect)
