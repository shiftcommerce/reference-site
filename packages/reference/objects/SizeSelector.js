// Libraries
import { Component } from 'react'

class SizeSelector extends Component {
  constructor (props) {
    super(props)
    this.renderOptions = this.renderOptions.bind(this)
  }

  renderOptions (variants) {
    return (
      variants && variants.map((variant, idx) =>
        <option role='option' key={idx} value={variant.id} aria-setsize={variants.length} aria-posinset={idx + 1}>{ variant.size }</option>
      )
    )
  }

  render () {
    let {
      name,
      prompt,
      variants,
      ...otherProps
    } = this.props

    return (
      <select name={name} {...otherProps}>

        <option role='option' value='' aria-setsize={variants.length} aria-posinset='0' >{ prompt }</option>

        { this.renderOptions(variants) }

      </select>
    )
  }
}

export default SizeSelector
