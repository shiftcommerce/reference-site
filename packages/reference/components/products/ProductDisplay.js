// Libraries
import { Component } from 'react'

// Components
import RelatedProducts from './RelatedProducts'
import DeliveryInformation from './DeliveryInformation'

// Objects
import ContentImage from '../../objects/ContentImage'
import Button from '../../objects/Button'
import DropdownSelect from '../../objects/DropdownSelect'
import SizeSelector from '../../objects/SizeSelector'

class ProductDisplay extends Component {
  render () {
    let {
      product
    } = this.props

    const quantityOptions = [ 1, 2, 3, 4, 5 ]

    return (
      <div className='c-product-display'>
        <div className='c-product-display__body'>
          <div className='c-product-display__gallery'>
            <div>
              <ContentImage src={product.asset_files[0] && product.asset_files[0].url} alt={product.title} height='705' width='503' aria-label={product.title} />
            </div>
          </div>

          <div className='c-product-display__details'>
            <section className='c-product-display__section'>
              <h1 role='heading' aria-level='1'>
                { product.title }
              </h1>

              <div className='c-product-display__sku'>
                <span>{ product.sku }</span>
              </div>

              <div className='c-product-display__price'>
                { product.price }
              </div>

              <div className='c-product-display__colour'>
                <strong>Colour:</strong>
                <span> { product.meta.colour } </span>
              </div>

              <div className='c-product-display__size'>
                <SizeSelector name='line_item[item_id]' prompt='Select a Size' variants={product.variants} aria-label='Size Selector' />
              </div>

              <div className='c-product-display__quantity'>
                <DropdownSelect name='line_item[unit_quantity]' prompt='Select a Quantity' options={quantityOptions} aria-label='Quantity Selector' />
              </div>

              <div className='c-product-display__add-button'>
                <Button label='ADD TO BAG' status='positive' aria-label='Add to Bag' />
              </div>
            </section>

            <section className='c-product-display__section' aria-label='Product Information'>
              <h3>Product Information</h3>
              { product.description }
            </section>

            <section className='c-product-display__section' aria-label='Delivery Information'>
              <DeliveryInformation />
            </section>

            <section className='c-product-display__section' aria-label='Related Products'>
              <RelatedProducts bundles={product.bundles} />
            </section>
          </div>
        </div>
      </div>
    )
  }
}

export default ProductDisplay
