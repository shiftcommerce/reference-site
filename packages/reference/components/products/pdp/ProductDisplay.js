// Libraries
import { Component } from 'react'
import classNames from 'classnames'
import Link from 'next/link'

// Components
import Carousel from './Carousel'
import ProductNavBar from '../../navigation/ProductNavBar'
import ProductPrice from './ProductPrice'

// Objects
import Button from '../../../objects/Button'
import SizeSelector from '../../../objects/SizeSelector'

// Fixtures
import Images from '../../../spec/fixtures/dummyPDPimages.fixture'

class ProductDisplay extends Component {
  renderRatingStars (product) {
    const rating = (product.rating || 0)

    return (
      [1, 2, 3, 4, 5].map((key) =>
        <span key={key} className={classNames({ 'c-product-display__info-rating--fill': key <= rating, 'c-product-display__info-rating--empty': key > rating })}>
          &#9733;
        </span>
      )
    )
  }

  renderCarousel () {
    return <>
      <div className='c-product-display__gallery'>
        <div className='c-carousel'>
          <Carousel className='c-carousel-slider' product={this.props.product} />
        </div>
      </div>
    </>
  }

  renderButtons () {
    const { product, addToBag } = this.props

    return <>
      <div className='c-product-display__buttons'>
        <div className='c-product-display__buttons-section1'>
          <div className='c-product-display__buttons-title'>
            {product.title}
          </div>
          <div className='c-product-display__buttons-price'>
            <ProductPrice variants={product.variants} />
          </div>
        </div>
        <div className='c-product-display__buttons-section2'>
          <div className='c-product-display__buttons-basket'>
            <Button className='c-product-display__buttons-basket-icon' label='ADD TO BASKET' status='primary' size='lrg' aria-label='Add to Basket' onClick={addToBag} />
          </div>
          <div className='c-product-display__buttons-buy'>
            <Button className='c-product-display__buttons-buy-icon' label='BUY' status='secondary' size='lrg' aria-label='Buy' />
          </div>
        </div>
      </div>
    </>
  }

  renderDescription () {
    const { product } = this.props

    return <>
      <div className='c-product-display__description'>
        <h1 className='c-product-display__description-title'>Product Details</h1>
        <label htmlFor='description' className='c-product-dispay__label' />
        <input type='checkbox' id='description' />
        <span className='c-product-display__arrow' />
        <div className='c-product-display__description-text' dangerouslySetInnerHTML={{ __html: product.description }} />
      </div>
    </>
  }

  renderInfo () {
    const { product, changeSize, sku } = this.props
    const productColour = product.meta_attributes.master_colour.value

    return <>
      <div className='c-product-display__info'>
        <div className='c-product-display__info-title'>
          {product.title}
        </div>
        <div className='c-product-display__info-price'>
          <ProductPrice variants={product.variants} />
        </div>
        <div className='c-product-display__info-rating'>
          { this.renderRatingStars(product) }
        </div>
        <p className='c-product-display__info-reviews'>Read Reviews</p>
        <div className='c-product-display__info-colour'>
          <h1 className='c-product-display__info-colour-title'>Colour</h1>
          <input type='radio' id='colour' style={{ backgroundColor: productColour }} />
          <label htmlFor='colour'>{productColour}</label>
        </div>
        <p className='c-product-display__info-size-chart'>Find your recommended size</p>
        <div className='c-product-display__info-size'>
          <SizeSelector onChange={changeSize} value={sku} name='line_item[item_id]' prompt='Please Select' label='Size' variants={product.variants} aria-label='Size Selector' />
        </div>
      </div>
    </>
  }

  renderFit (meta) {
    if (meta.silhouette) {
      return <>
        <p className='c-product-display__size-text'>- {meta.silhouette.value}</p>
      </>
    }
  }

  renderFeature (meta) {
    if (meta.product_feature) {
      return (
        <p className='c-product-display__size-text'>- {meta.product_feature.value}</p>
      )
    }
  }

  renderSizeAndFit () {
    const { product: { meta_attributes } } = this.props

    if (meta_attributes.product_feature || meta_attributes.silhouette) {
      return <>
        <div className='c-product-display__size'>
          <h1 className='c-product-display__size-title'>Size & Fit</h1>
          <label htmlFor='size' className='c-product-dispay__label' />
          <input type='checkbox' id='size' />
          <span className='c-product-display__arrow' />
          { this.renderFeature(meta_attributes) }
          { this.renderFit(meta_attributes) }
        </div>
      </>
    }
  }

  renderDetailsAndCare () {
    const { product } = this.props
    const cleaning = product.meta_attributes.care_instructions.value
    const fabric = product.meta_attributes.fabric.value

    return <>
      <div className='c-product-display__care'>
        <h1 className='c-product-display__care-title'>Details & Care</h1>
        <label htmlFor='care' className='c-product-dispay__label' />
        <input type='checkbox' id='care' />
        <span className='c-product-display__arrow' />
        <p className='c-product-display__care-text'>- {fabric}</p>
        <p className='c-product-display__care-text'>- {cleaning}</p>
      </div>
    </>
  }

  // Currently using dummy data from fixture
  renderRelatedProducts () {
    return <>
      <div className='c-product-display__related'>
        <h1 className='c-product-display__related-title'><span>you may also like</span></h1>
        <div className='c-product-display__related-images'>
          { Images.asset_files.map((image, index) => {
            return (
              <Link href={image.link} key={index}>
                <div className='c-product-display__related-images-link'>
                  <img className='c-product-display__related-images-file' src={image.asset_file} />
                  <p className='c-product-display__related-images-text'>{image.text}</p>
                  <p className='c-product-display__related-images-price'>{image.price}</p>
                </div>
              </Link>
            )
          })
        }
        </div>
      </div>
    </>
  }

  render () {
    return <>
      <div className='c-product-display'>
        <ProductNavBar />
        <div className='c-product-display__body'>
          { this.renderCarousel() }
          { this.renderButtons() }
          { this.renderInfo() }
          { this.renderDescription() }
          { this.renderSizeAndFit() }
          { this.renderDetailsAndCare() }
          { this.renderRelatedProducts() }
        </div>
      </div>
    </>
  }
}

export default ProductDisplay
