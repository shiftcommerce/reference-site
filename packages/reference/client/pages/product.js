// Libraries
import { Component } from 'react'
import { connect } from 'react-redux'
import Head from 'next/head'

// Lib
import renderComponents from '../lib/render-components'
import { algoliaReduxWrapper } from '../lib/algolia-redux-wrapper'

// Components
import Loading from '../components/loading'
import ProductDisplay from '../components/products/display/product-display'

// Actions
import { addToCart } from '../actions/cart-actions'
import { readProduct } from '../actions/product-actions'

export class Product extends Component {
  constructor (props) {
    super(props)
    // TODO: Update to use redux
    this.state = {
      sku: '',
      variant: '',
      variantId: null,
      quantity: 1,
      price: '',
      stockAvailableLevel: 0
    }

    this.changeVariant = this.changeVariant.bind(this)
    this.addToBag = this.addToBag.bind(this)
  }

  static async getInitialProps ({ reduxStore, req, query }) {
    const { id } = query
    const isServer = !!req

    if (isServer) {
      await reduxStore.dispatch(readProduct(id))
    }

    return { id: id }
  }

  componentDidMount () {
    const { dispatch, id } = this.props
    dispatch(readProduct(id))
  }

  addToBag () {
    const { variantId, quantity } = this.state
    this.props.dispatch(addToCart(variantId, parseInt(quantity)))
  }

  changeVariant (e) {
    this.setState({
      stockAvailableLevel: parseInt(e.target.attributes['data-stock-available-level'].value, 10),
      sku: e.target.value,
      variant: e.target.text,
      variantId: e.target.attributes['data-variant-id'].value,
      price: e.target.attributes['data-price'].value
    })
  }

  render () {
    const { product, product: { loading, error, template, title } } = this.props

    if (loading) {
      return (
        <Loading />
      )
    } else if (error || Object.keys(product).length === 0) {
      return (
        <h1>Unable to load product.</h1>
      )
    } else {
      const templateSection = template && template.sections && template.sections.slice(-1).pop()
      const components = templateSection && templateSection.components
      const selectedVariant = product.variants.find(variant => variant.id === this.state.variantId)

      return (
        <>
          <Head>
            <title>{ title }</title>
          </Head>
          <ProductDisplay
            product={product}
            changeVariant={this.changeVariant}
            addToBag={this.addToBag}
            clickToBuy={this.clickToBuy}
            selectedVariant={selectedVariant}
          />
          { components && renderComponents(components) }
        </>
      )
    }
  }
}

function mapStateToProps (state) {
  const { menu, product, cart } = state
  return { menu, product, cart }
}

export default algoliaReduxWrapper(connect(mapStateToProps)(Product), Product)
