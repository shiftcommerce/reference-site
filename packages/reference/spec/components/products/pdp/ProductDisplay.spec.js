// Component
import ProductDisplay from '../../../../components/products/pdp/ProductDisplay'
import DeliveryInformation from '../../../../components/products/pdp/DeliveryInformation'
import Carousel from '../../../../components/products/pdp/Carousel'
// Objects
import Button from '../../../../objects/Button'
import DropdownSelect from '../../../../objects/DropdownSelect'
import SizeSelector from '../../../../objects/SizeSelector'

// Fixtures
import product from '../../../fixtures/product.fixture'

test('renders correctly', () => {
  // Arrange
  const quantityOptions = [ 1, 2, 3, 4, 5 ]
  const emptyFunction = () => {}
  const value = {
    sku: '',
    size: 'Small',
    quantity: ''
  }

  // Act
  const wrapper = mount(
    <ProductDisplay product={product} changeQuantity={emptyFunction} changeSize={emptyFunction} addToBag={emptyFunction} {...value} />
  )

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toIncludeText(product.title)
  expect(wrapper).toIncludeText(product.description)
  expect(wrapper).toIncludeText(product.variants[0].sku)
  expect(wrapper).toContainReact(<Carousel className='c-carousel-slider' product={product} />)
  expect(wrapper).toContainReact(<SizeSelector onChange={emptyFunction} value='' name='line_item[item_id]' prompt='Select a Size' variants={product.variants} aria-label='Size Selector' />)
  expect(wrapper).toContainReact(<DropdownSelect onChange={emptyFunction} value='' name='line_item[unit_quantity]' prompt='Select a Quantity' options={quantityOptions} aria-label='Quantity Selector' />)
  expect(wrapper).toContainReact(<Button label='ADD TO BAG' status='primary' size='lrg' aria-label='Add to Bag' onClick={emptyFunction} />)
  expect(wrapper).toContainReact(<DeliveryInformation />)
})
