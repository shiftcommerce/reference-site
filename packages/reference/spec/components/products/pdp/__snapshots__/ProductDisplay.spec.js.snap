// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ProductDisplay
  addToBag={[Function]}
  changeQuantity={[Function]}
  changeSize={[Function]}
  product={
    Object {
      "asset_files": Array [
        Object {
          "alt_text": "Textured Long T-Shirt",
          "asset_file": "S2623408_C29P_Alt2.jpg",
          "s3_url": "https://staging-matalanintegration-1452506760.s3.amazonaws.com/uploads/asset_file/asset_file/12671/S2623408_C29P_Alt2.jpg",
        },
      ],
      "bundles": Array [
        Object {
          "id": 1,
          "products": Array [
            Object {
              "asset_files": Array [
                Object {
                  "id": 2,
                  "url": "https://staging-matalanintegration-1452506760.s3.amazonaws.com/uploads/asset_file/asset_file/12917/S2623404_C146_Main.jpg",
                },
              ],
              "canonical_path": "/products/stripe-shirt",
              "id": 1,
              "title": "Stripe Shirt",
            },
            Object {
              "asset_files": Array [],
              "canonical_path": "/products/slip-on-pump",
              "id": 2,
              "title": "Slip on Pump (no asset files example)",
            },
          ],
          "title": "Mingo",
        },
      ],
      "canonical_path": "1",
      "description": "Part of our Made in Britain collection, the limited edition Thruxton jacket by Morley offers great style and practicality for everyday wear.",
      "id": 1,
      "meta_attributes": Object {
        "master_colour": Object {
          "value": "Stone",
        },
      },
      "price": "99.00",
      "title": "Textured Long T-Shirt",
      "variants": Array [
        Object {
          "meta_attributes": Object {
            "size": Object {
              "value": "Size 8",
            },
          },
          "price": "10.00",
          "sku": 1,
          "title": "Small",
        },
        Object {
          "meta_attributes": Object {
            "size": Object {
              "value": "Size 10",
            },
          },
          "price": "11.00",
          "sku": 2,
          "title": "Medium",
        },
        Object {
          "meta_attributes": Object {
            "size": Object {
              "value": "Size 12",
            },
          },
          "price": "12.00",
          "sku": 3,
          "title": "Large",
        },
      ],
    }
  }
  quantity=""
  size="Small"
  sku=""
>
  <div
    className="c-product-display"
  >
    <div
      className="c-product-display__body"
    >
      <div
        className="c-product-display__gallery"
      >
        <div
          className="c-carousel"
        >
          <Carousel
            className="c-carousel-slider"
            product={
              Object {
                "asset_files": Array [
                  Object {
                    "alt_text": "Textured Long T-Shirt",
                    "asset_file": "S2623408_C29P_Alt2.jpg",
                    "s3_url": "https://staging-matalanintegration-1452506760.s3.amazonaws.com/uploads/asset_file/asset_file/12671/S2623408_C29P_Alt2.jpg",
                  },
                ],
                "bundles": Array [
                  Object {
                    "id": 1,
                    "products": Array [
                      Object {
                        "asset_files": Array [
                          Object {
                            "id": 2,
                            "url": "https://staging-matalanintegration-1452506760.s3.amazonaws.com/uploads/asset_file/asset_file/12917/S2623404_C146_Main.jpg",
                          },
                        ],
                        "canonical_path": "/products/stripe-shirt",
                        "id": 1,
                        "title": "Stripe Shirt",
                      },
                      Object {
                        "asset_files": Array [],
                        "canonical_path": "/products/slip-on-pump",
                        "id": 2,
                        "title": "Slip on Pump (no asset files example)",
                      },
                    ],
                    "title": "Mingo",
                  },
                ],
                "canonical_path": "1",
                "description": "Part of our Made in Britain collection, the limited edition Thruxton jacket by Morley offers great style and practicality for everyday wear.",
                "id": 1,
                "meta_attributes": Object {
                  "master_colour": Object {
                    "value": "Stone",
                  },
                },
                "price": "99.00",
                "title": "Textured Long T-Shirt",
                "variants": Array [
                  Object {
                    "meta_attributes": Object {
                      "size": Object {
                        "value": "Size 8",
                      },
                    },
                    "price": "10.00",
                    "sku": 1,
                    "title": "Small",
                  },
                  Object {
                    "meta_attributes": Object {
                      "size": Object {
                        "value": "Size 10",
                      },
                    },
                    "price": "11.00",
                    "sku": 2,
                    "title": "Medium",
                  },
                  Object {
                    "meta_attributes": Object {
                      "size": Object {
                        "value": "Size 12",
                      },
                    },
                    "price": "12.00",
                    "sku": 3,
                    "title": "Large",
                  },
                ],
              }
            }
          >
            <div
              className="c-carousel__content"
            >
              <Slider
                arrows={true}
                dots={true}
                fade={true}
                infinite={true}
                slidesToScroll={1}
                slidesToShow={1}
                speed={500}
              >
                <InnerSlider
                  accessibility={true}
                  adaptiveHeight={false}
                  afterChange={null}
                  appendDots={[Function]}
                  arrows={true}
                  autoplay={false}
                  autoplaySpeed={3000}
                  beforeChange={null}
                  centerMode={false}
                  centerPadding="50px"
                  className=""
                  cssEase="ease"
                  customPaging={[Function]}
                  dots={true}
                  dotsClass="slick-dots"
                  draggable={true}
                  easing="linear"
                  edgeFriction={0.35}
                  fade={true}
                  focusOnSelect={false}
                  infinite={true}
                  initialSlide={0}
                  lazyLoad={null}
                  nextArrow={null}
                  onEdge={null}
                  onInit={null}
                  onLazyLoadError={null}
                  onReInit={null}
                  pauseOnDotsHover={false}
                  pauseOnFocus={false}
                  pauseOnHover={true}
                  prevArrow={null}
                  responsive={null}
                  rows={1}
                  rtl={false}
                  slide="div"
                  slidesPerRow={1}
                  slidesToScroll={1}
                  slidesToShow={1}
                  speed={500}
                  swipe={true}
                  swipeEvent={null}
                  swipeToSlide={false}
                  touchMove={true}
                  touchThreshold={5}
                  unslick={true}
                  useCSS={true}
                  useTransform={true}
                  variableWidth={false}
                  vertical={false}
                  waitForAnimate={true}
                >
                  <div
                    className="slick-slider slick-initialized"
                  >
                    <div
                      className="slick-list"
                    >
                      <Track
                        centerMode={false}
                        centerPadding="50px"
                        cssEase="ease"
                        currentSlide={0}
                        fade={true}
                        focusOnSelect={null}
                        infinite={true}
                        lazyLoad={null}
                        lazyLoadedList={Array []}
                        listHeight={0}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        onMouseOver={[Function]}
                        rtl={false}
                        slideCount={1}
                        slideHeight={0}
                        slideWidth={0}
                        slidesToScroll={1}
                        slidesToShow={1}
                        speed={500}
                        trackStyle={
                          Object {
                            "opacity": 1,
                          }
                        }
                        unslick={true}
                        variableWidth={false}
                        vertical={false}
                      >
                        <div
                          className="slick-track"
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          onMouseOver={[Function]}
                          style={
                            Object {
                              "opacity": 1,
                            }
                          }
                        >
                          <div
                            aria-hidden={false}
                            className="slick-slide slick-active slick-current"
                            data-index={0}
                            key="original0"
                            onClick={[Function]}
                            style={
                              Object {
                                "WebkitTransition": "opacity 500ms ease, visibility 500ms ease",
                                "left": -0,
                                "opacity": 1,
                                "outline": "none",
                                "position": "relative",
                                "transition": "opacity 500ms ease, visibility 500ms ease",
                                "width": 0,
                              }
                            }
                            tabIndex="-1"
                          >
                            <div
                              key="0"
                            >
                              <div
                                key="0"
                                style={
                                  Object {
                                    "display": "inline-block",
                                    "width": "100%",
                                  }
                                }
                                tabIndex={-1}
                              >
                                <Image
                                  className="c-carousel__image"
                                  src="https://staging-matalanintegration-1452506760.s3.amazonaws.com/uploads/asset_file/asset_file/12671/S2623408_C29P_Alt2.jpg"
                                >
                                  <div
                                    className="c-carousel__image"
                                  >
                                    <LazyLoad
                                      height={200}
                                      offset={0}
                                      once={false}
                                      overflow={false}
                                      resize={false}
                                      scroll={true}
                                      unmountIfInvisible={false}
                                    >
                                      <div
                                        className="lazyload-placeholder"
                                        style={
                                          Object {
                                            "height": 200,
                                          }
                                        }
                                      />
                                    </LazyLoad>
                                  </div>
                                </Image>
                              </div>
                            </div>
                          </div>
                        </div>
                      </Track>
                    </div>
                  </div>
                </InnerSlider>
              </Slider>
            </div>
          </Carousel>
        </div>
      </div>
      <div
        className="c-product-display__details"
      >
        <section
          className="c-product-display__section"
        >
          <h1
            aria-level="1"
            role="heading"
          >
            Textured Long T-Shirt
          </h1>
          <div
            className="c-product-display__sku"
          >
            <span />
          </div>
          <div
            className="c-product-display__price"
          >
            <ProductPrice
              variants={
                Array [
                  Object {
                    "meta_attributes": Object {
                      "size": Object {
                        "value": "Size 8",
                      },
                    },
                    "price": "10.00",
                    "sku": 1,
                    "title": "Small",
                  },
                  Object {
                    "meta_attributes": Object {
                      "size": Object {
                        "value": "Size 10",
                      },
                    },
                    "price": "11.00",
                    "sku": 2,
                    "title": "Medium",
                  },
                  Object {
                    "meta_attributes": Object {
                      "size": Object {
                        "value": "Size 12",
                      },
                    },
                    "price": "12.00",
                    "sku": 3,
                    "title": "Large",
                  },
                ]
              }
            >
              <div
                className="c-product-price"
              >
                <span>
                  £
                  10.00
                   - £
                  12.00
                </span>
              </div>
            </ProductPrice>
          </div>
          <div
            className="c-product-display__description"
            dangerouslySetInnerHTML={
              Object {
                "__html": "Part of our Made in Britain collection, the limited edition Thruxton jacket by Morley offers great style and practicality for everyday wear.",
              }
            }
          />
          <div
            className="c-product-display__colour"
          >
            <strong>
              Colour:
            </strong>
            <span>
               
              Stone
               
            </span>
          </div>
          <div
            className="c-product-display__size"
          >
            <SizeSelector
              aria-label="Size Selector"
              name="line_item[item_id]"
              onChange={[Function]}
              prompt="Select a Size"
              value=""
              variants={
                Array [
                  Object {
                    "meta_attributes": Object {
                      "size": Object {
                        "value": "Size 8",
                      },
                    },
                    "price": "10.00",
                    "sku": 1,
                    "title": "Small",
                  },
                  Object {
                    "meta_attributes": Object {
                      "size": Object {
                        "value": "Size 10",
                      },
                    },
                    "price": "11.00",
                    "sku": 2,
                    "title": "Medium",
                  },
                  Object {
                    "meta_attributes": Object {
                      "size": Object {
                        "value": "Size 12",
                      },
                    },
                    "price": "12.00",
                    "sku": 3,
                    "title": "Large",
                  },
                ]
              }
            >
              <select
                aria-label="Size Selector"
                name="line_item[item_id]"
                onChange={[Function]}
                value=""
              >
                <option
                  aria-posinset="0"
                  aria-setsize={3}
                  role="option"
                  value=""
                >
                  Select a Size
                </option>
                <option
                  aria-posinset={1}
                  aria-setsize={3}
                  data-price="10.00"
                  key="0"
                  role="option"
                  value={1}
                >
                  Size 8
                </option>
                <option
                  aria-posinset={2}
                  aria-setsize={3}
                  data-price="11.00"
                  key="1"
                  role="option"
                  value={2}
                >
                  Size 10
                </option>
                <option
                  aria-posinset={3}
                  aria-setsize={3}
                  data-price="12.00"
                  key="2"
                  role="option"
                  value={3}
                >
                  Size 12
                </option>
              </select>
            </SizeSelector>
          </div>
          <div
            className="c-product-display__quantity"
          >
            <_class
              aria-label="Quantity Selector"
              name="line_item[unit_quantity]"
              onChange={[Function]}
              options={
                Array [
                  1,
                  2,
                  3,
                  4,
                  5,
                ]
              }
              prompt="Select a Quantity"
              value=""
            >
              <DropdownSelect
                aria-label="Quantity Selector"
                name="line_item[unit_quantity]"
                onChange={[Function]}
                options={
                  Array [
                    1,
                    2,
                    3,
                    4,
                    5,
                  ]
                }
                prompt="Select a Quantity"
                renderValidationMessage={[Function]}
                value=""
              >
                <div
                  className="o-form__input-group"
                >
                  <select
                    className="o-form__input-block o-form__input-field"
                    id="line_item[unit_quantity]"
                    name="line_item[unit_quantity]"
                    onBlur={[Function]}
                    onChange={[Function]}
                    role="listbox"
                    value=""
                  >
                    <option
                      aria-posinset="0"
                      aria-setsize={5}
                      role="option"
                      value=""
                    >
                      Select a Quantity
                    </option>
                    <option
                      aria-posinset={1}
                      aria-setsize={5}
                      key="0"
                      role="option"
                      value={1}
                    >
                      1
                    </option>
                    <option
                      aria-posinset={2}
                      aria-setsize={5}
                      key="1"
                      role="option"
                      value={2}
                    >
                      2
                    </option>
                    <option
                      aria-posinset={3}
                      aria-setsize={5}
                      key="2"
                      role="option"
                      value={3}
                    >
                      3
                    </option>
                    <option
                      aria-posinset={4}
                      aria-setsize={5}
                      key="3"
                      role="option"
                      value={4}
                    >
                      4
                    </option>
                    <option
                      aria-posinset={5}
                      aria-setsize={5}
                      key="4"
                      role="option"
                      value={5}
                    >
                      5
                    </option>
                  </select>
                </div>
              </DropdownSelect>
            </_class>
          </div>
          <div
            className="c-product-display__add-button"
          >
            <Button
              aria-label="Add to Bag"
              label="ADD TO BAG"
              onClick={[Function]}
              size="lrg"
              status="primary"
            >
              <button
                aria-label="Add to Bag"
                className="o-button o-button--primary o-button--lrg"
                disabled={false}
                onClick={[Function]}
                role="button"
                tabIndex="0"
              >
                ADD TO BAG
              </button>
            </Button>
          </div>
        </section>
        <section
          aria-label="Delivery Information"
          className="c-product-display__section"
        >
          <DeliveryInformation>
            <div>
              <h3>
                Delivery & Returns
              </h3>
              <p>
                Click & collect for free from your local store in 5 working days. Delivery hours are between 8am-8pm, Monday-Friday.
              </p>
              <p>
                Standard delivery will be charged at £3.95 regardless of how many items you order.
              </p>
              <p>
                Next working day delivery will be charged at £4.95 regardless of how many items you order. Your order will be with you the next working day if you order before 4pm. Any orders placed after 4pm Thursday up to 4pm Sunday will be delivered Monday. Please note our Next Day Delivery service is not available on selected items due to their size or weight.
              </p>
            </div>
          </DeliveryInformation>
        </section>
      </div>
    </div>
  </div>
</ProductDisplay>
`;
